From 2bca7670971b25bf83605b89469e9b8195e73860 Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Mon, 28 Apr 2014 15:49:26 +0200
Subject: [LIBREPORT PATCH 07/10] hide "Don't store password" checkbox

Show the checkbox only if the options can be stored. For example, it does
not make sense to show this checkbox when in Anaconda.

Related to #259

Signed-off-by: Jakub Filak <jfilak@redhat.com>
---
 src/gtk-helpers/event_config_dialog.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/gtk-helpers/event_config_dialog.c b/src/gtk-helpers/event_config_dialog.c
index 1ed5196..ff3e38b 100644
--- a/src/gtk-helpers/event_config_dialog.c
+++ b/src/gtk-helpers/event_config_dialog.c
@@ -237,7 +237,10 @@ config_dialog_t *create_event_config_dialog_content(event_config_t *event, GtkWi
     g_list_foreach(event->options, &add_option_to_table, option_table);
 
     /* if there is at least one password option, add checkbox to disable storing passwords */
-    if (has_password_option)
+    /* if the user storage is not available nothing is to be stored, so it is not necessary
+     * to bother with an extra checkbox about storing passwords */
+    if (is_event_config_user_storage_available()
+            && has_password_option)
     {
         unsigned last_row = add_one_row_to_grid(GTK_GRID(option_table));
         GtkWidget *pass_store_cb = gtk_check_button_new_with_label(_("Don't store passwords"));
-- 
1.8.3.1

